src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -169,20 +169,31 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
 #endif
     }
 
+    bool haveCompletedDeferredInit{false};
     bool init() noexcept override
     {
-#if JUCE_LINUX
-        if (_host.canUseTimerSupport())
-        {
-            _host.timerSupportRegister(1000 / 50, &idleTimer);
-        }
-#endif
         defineAudioPorts();
 
+        haveCompletedDeferredInit = false;
+        _host.requestCallback();
```

Comment:
Just to make sure I'm understanding correctly: The idea is that if `init()` is called on a thread that isn't the main thread, defer the timer initialization to a main thread callback?

Reply:
A bit different. The idea is that in init the host doesn’t know if a plugin supports timers. So it’s not clear that we can call rather than just probe extensions.  See https://github.com/free-audio/clap/discussions/111#discussioncomment-3007976